### PR TITLE
[Feature] Support column rename for olap table Part 1

### PR DESCRIPTION
**Pull request by Astralidea**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/catalog/Column.java [hunk 1]
```diff
@@ -80,6 +80,13 @@ public class Column implements Writable, GsonPreProcessable, GsonPostProcessable
 
     @SerializedName(value = "name")
     private String name;
+
+    // If the logical name exists,
+    // the logical name will be used and converted to the physical name when the user uses it.
+    // Use physical name if not present
+    @SerializedName("logicalName")
+    private String logicalName;
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Column.java:88]: Better to add comments on the relationship between logicalName and name in front of this class.

File: fe/fe-core/src/main/java/com/starrocks/catalog/Column.java [hunk 2]
```diff
@@ -244,6 +251,21 @@ public String getName() {
...
+            return name;
+        }
+        return logicalName;
+    }
+
+    public void setLogicalName(String logicalName) {
+        if (name.equalsIgnoreCase(logicalName)) {
+            this.logicalName = null;
+            return;
+        }
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Column.java:265]: It is weird to compare to name when setting `logicalName`

**Review by imay — COMMENTED** [review]
Actually, I'm wondering why not add a new field of PhysicalColumn in Column.java?
Then origin column will use name as PhysicalColumn?
For Frontend, column name is treated as logical name in most cases. So if we keep column as logical name, most client code doesn't need to change.

**Comment by Astralidea** [comment]
> Actually, I'm wondering why not add a new field of PhysicalColumn in Column.java? Then origin column will use name as PhysicalColumn? For Frontend, column name is treated as logical name in most cases. So if we keep column as logical name, most client code doesn't need to change.

You are right, I wrote another version using your idea, and can achieve the same testing function by changing half the files. This closure is done on top of the other.

**Pull request closed without merging**